src/services/pcn-packetcapture/src/Packetcapture.cpp

```diff
@@ -435,4 +376,65 @@ void Packetcapture::to_timeval(std::chrono::system_clock::duration& d, struct ti
 
     tv.tv_sec  = sec.count();
     tv.tv_usec = std::chrono::duration_cast<std::chrono::microseconds>(d - sec).count();
+}
+
+void Packetcapture::setFilter(const std::string &value) {
+    this->filter = value;
+    logger()->info("Inserted filter: {0}", filter);
+    if (filter == "all") {
+      filterCode = "return pcn_pkt_controller(ctx, md, reason);";
+    }else {
+        memset(&cbpf, 0, sizeof(cbpf));
+        filterCompile(filter, &cbpf);
+    }
+    updateFilter();
+}
+
+std::string Packetcapture::getFilter() {
+  return this->filter;
+}
+
+uint32_t Packetcapture::getSnaplen() {
+  return this->snaplen;
```

Comment:
why did you introduce an additional snaplen value and not use the globalheader class?

Reply:
thanks, I've changed, now I use the globalheader class